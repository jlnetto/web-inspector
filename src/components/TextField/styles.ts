import styled from 'styled-components';

export const StyledTextField = styled.input`
  flex: 1;
  padding: 10px;
  font-size: 14px;
  border: 1px solid #ccc;
  border-radius: 4px;

  &:focus {
    outline: none !important;
    border: 2px solid #007bff;
  }
`;

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  gap: 5px;
`;
