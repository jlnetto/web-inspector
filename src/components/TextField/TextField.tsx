import { StyledTextField, Wrapper } from './styles';

type InputProps = {
  name: string;
  label: string;
  value: string;
  onChange: (value: string) => void;
  placeholder?: string;
};

const TextField: React.FC<InputProps> = ({
  name,
  label,
  onChange,
  placeholder,
  ...rest
}) => {
  return (
    <Wrapper>
      <label htmlFor={name}>{label}</label>
      <StyledTextField
        {...rest}
        onChange={(e) => onChange(e.target.value)}
        placeholder={placeholder}
      />
    </Wrapper>
  );
};

export default TextField;
