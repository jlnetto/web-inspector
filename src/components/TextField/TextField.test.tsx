import { render, fireEvent } from '@testing-library/react';
import TextField from './TextField';

const onChangeMock = jest.fn();
describe('TextField component', () => {
  it('render input correctly', () => {
    const { asFragment } = render(
      <TextField
        name="username"
        label="Username"
        onChange={onChangeMock}
        value=""
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('calls onChange with input value', () => {
    const { getByRole } = render(
      <TextField
        name="username"
        label="Username"
        onChange={onChangeMock}
        value=""
      />
    );

    const input = getByRole('textbox');
    fireEvent.change(input, { target: { value: 'testuser' } });

    expect(onChangeMock).toHaveBeenCalledWith('testuser');
  });
});
