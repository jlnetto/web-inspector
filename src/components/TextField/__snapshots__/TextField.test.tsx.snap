// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField component render input correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-gTRrQi gWFUFE"
  >
    <label
      for="username"
    >
      Username
    </label>
    <input
      class="sc-bgqQcB eEhfAl"
      value=""
    />
  </div>
</DocumentFragment>
`;
